code/modules/projectiles/guns/projectile/pistol.dm

```diff
@@ -86,4 +86,36 @@
 	icon_state = "[initial(icon_state)][silenced ? "-silencer" : ""][chambered ? "" : "-e"]"
 	return
 
-
+/obj/item/weapon/gun/projectile/handgun //mime fingergun 
+	name = "hand-gun"
+	desc = "This is a stickup!"
+	icon_state = "handgun"
+	inhand_states = null
+	ammo_type = "/obj/item/ammo_casing/invisible"
+	mag_type = "/obj/item/ammo_storage/magazine/invisible"
+	cant_drop = 1
+	gun_flags = 0
+	silenced = 1
+	fire_sound = null
+	load_method = 2
+	var/obj/item/clothing/gloves/white/advanced/linked_gloves = null
```

Comment:
I'm not seeing the point of this var. It holds a reference to the gloves until it nulls them in Destroy(), but the gloves should really be handling that instead.
If you take it out, we'd be able to use the invisible gun with other things, like a spell.